gammapy/modeling/models/spatial.py

```diff
@@ -604,13 +608,15 @@ def read(cls, filename, normalize=True, **kwargs):
         m = Map.read(filename, **kwargs)
         if m.unit == "":
             m.unit = "sr-1"
+        elif m.unit != "sr-1":
+            raise ValueError("The map unit should be in sr-1")
         return cls(m, normalize=normalize, filename=filename)
 
     def evaluate(self, lon, lat, norm):
         """Evaluate model."""
         coord = {"lon": lon.to_value("deg"), "lat": lat.to_value("deg")}
         val = self.map.interp_by_coord(coord, **self._interp_kwargs)
-        return u.Quantity(norm.value * val, self.map.unit, copy=False)
+        return u.Quantity(norm.value * val, "sr-1", copy=False)
```

Comment:
This some is basically redundant. Once you have checked the map unit on init you can just use it here and it's guaranteed to be compatible...